### PR TITLE
Fix: improve logging on cluster create attempt, cleanup

### PR DESCRIPTION
**Pull request by 678098**
Following the internal ticket 175315279
Changes:
- Improve logging when no cluster found in the cluster catalog
- More precondition checks
- Remove `int` return code and `bsl::ostream &errorDescription` argument from `ClusterQueueHelper::initialize`, `ClusterStateMonitor::start` and `ClusterProxy::startDispatched` since these methods are trivial and in fact cannot return non-zero error code
- Early close `ifstream`s when we already read everything

### CODE REVIEW

File: src/applications/bmqbrkr/bmqbrkr.m.cpp [hunk 1]
```diff
@@ -274,6 +274,7 @@ static int getConfig(bsl::ostream&      errorDescription,
                          << "[file: " << configFilename << "]";
         return rc_GENERATION_FAILED;  // RETURN
     }
+    configStream.close();
```
Comment (678098) [inline src/applications/bmqbrkr/bmqbrkr.m.cpp:277]: We read everything from that file via `rdbuf()` before and can close it early

File: src/groups/mqb/mqbblp/mqbblp_cluster.cpp [hunk 2]
```diff
@@ -108,9 +108,7 @@ void Cluster::startDispatched(bsl::ostream* errorDescription, int* rc)
         rc_SUCCESS             = 0,
         rc_STATE_MGR_FAILURE   = -1,
         rc_MISC_FAILURE        = -2,
-        rc_STORAGE_MGR_FAILURE = -3,
-        rc_QUEUE_HELPER        = -4,
-        rc_STATE_MONITOR       = -5
```
Comment (678098) [inline src/groups/mqb/mqbblp/mqbblp_cluster.cpp:113]: We would never see these removed return codes for a very long time

File: src/groups/mqb/mqbblp/mqbblp_clustercatalog.cpp [hunk 3]
```diff
@@ -198,6 +198,8 @@ int ClusterCatalog::createCluster(bsl::ostream& errorDescription,
 
         if (proxyDefinitionIter ==
             d_clustersDefinition.proxyClusters().end()) {
+            errorDescription << "Fetch proxy definition failed for cluster: '"
+                             << name << "'";
```
Comment (678098) [inline src/groups/mqb/mqbblp/mqbblp_clustercatalog.cpp:202]: This error description was missing

File: src/groups/mqb/mqbblp/mqbblp_clustercatalog.cpp [hunk 4]
```diff
@@ -419,6 +425,7 @@ int ClusterCatalog::loadBrokerClusterConfig(bsl::ostream& errorDescription)
         return rc_CONFIG_READ_ERROR * 10 +
                rc_BROKER_CLUSTER_CONFIG_LOADFAILURE;  // RETURN
     }
+    configStream.close();
```
Comment (678098) [inline src/groups/mqb/mqbblp/mqbblp_clustercatalog.cpp:428]: We read everything from that file via `rdbuf()` before and can close it early

File: src/groups/mqb/mqbblp/mqbblp_clusterproxy.cpp [hunk 5]
```diff
@@ -120,22 +120,15 @@ void ClusterProxy::startDispatched(bsl::ostream* errorDescription, int* rc)
     BSLS_ASSERT_SAFE(!d_isStarted &&
                      "start() can only be called once on this object");
 
-    enum { rc_SUCCESS = 0, rc_ACTIVE_NODE_MANAGER = -1, rc_QUEUE_HELPER = -2 };
-
-    *rc = d_queueHelper.initialize(*errorDescription);
```
Comment (678098) [inline src/groups/mqb/mqbblp/mqbblp_clusterproxy.cpp:125]: See the `QueueHelper::initialize`, it cannot fail, so we can simplify calls

File: src/groups/mqb/mqbblp/mqbblp_queuesessionmanager.cpp [hunk 6]
```diff
@@ -195,7 +195,8 @@ void QueueSessionManager::onDomainOpenCb(
         // Failed to open domain
         BALL_LOG_WARN << "#CLIENT_OPENQUEUE_FAILURE "
                       << d_dispatcherClient_p->description()
-                      << ": Error while opening domain: " << status;
+                      << ": Error while opening domain: [reason: " << status
+                      << ", request: " << request << "]";
```
Comment (678098) [inline src/groups/mqb/mqbblp/mqbblp_queuesessionmanager.cpp:199]: Here, `request` was missing

File: src/groups/mqb/mqba/mqba_clientsession.cpp [hunk 7]
```diff
@@ -1289,8 +1289,8 @@ void ClientSession::openQueueCb(
         response.choice().makeStatus(status);
 
         BALL_LOG_WARN << "#CLIENT_OPENQUEUE_FAILURE " << description()
-                      << ": Error while opening queue: [reason: '" << status
-                      << "', request: " << handleParamsCtrlMsg << "]";
+                      << ": Error while opening queue: [reason: " << status
```
Comment (dorjesinpo) [inline src/groups/mqb/mqba/mqba_clientsession.cpp:1292]: the quote was there for a good reason -  `status` is a sentence

Comment (678098) [reply src/groups/mqb/mqba/mqba_clientsession.cpp:1292]: https://github.com/bloomberg/blazingmq/blob/21feef613674bbafe68f34ee5c849b837676a6bc/src/groups/mqb/mqba/mqba_clientsession.cpp#L1269

Here, `const bmqp_ctrlmsg::Status& status` is a schema object, so it's not a string, it's a structure with some fields.

An example output _before the PR_ is here:
`
WARN mqba_clientsession.cpp:1294 #CLIENT_OPENQUEUE_FAILURE testapp.tsk:89302: Error while opening queue: [reason: '[ category = E_UNKNOWN code = -1 message = "AppId should not be specified when opening a non-fanout queue" ]', request: [ rId = 2 choice = [ openQueue = [ handleParameters = [ uri = "bmq://test.mmap.priority?id=abd" qId = 0 subIdInfo = [ subId = 1 appId = "abd" ] flags = 2 readCount = 1 writeCount = 0 adminCount = 0 ] ] ] ]]
`

Comment (dorjesinpo) [reply src/groups/mqb/mqba/mqba_clientsession.cpp:1292]: Got it. Thanks

**Review by dorjesinpo — APPROVED** [review]

**Pull request merged**